sql/parser/eval_test.go

```diff
@@ -308,3 +315,25 @@ func TestEvalExprError(t *testing.T) {
 		}
 	}
 }
+
+func TestSimilarEscape(t *testing.T) {
+	testData := []struct {
+		expr     string
+		expected string
+	}{
+		{`test`, `test`},
+		{`test%`, `test.*`},
+		{`_test_`, `.test.`},
+		{`_%*`, `..**`},
+		{`[_%]*`, `[_%]*`},
+		{`.^`, `\.\^`},
+		{`%(b|d)%`, `.*(?:b|d).*`},
+		{`se\"arch\"[\"]`, `se(arch)[\"]`},
```

Comment:
Yes, I tested postgres via the `psql` tool. And yes, this is good for SIMILAR TO as well, though we might revisit at some point.